### PR TITLE
init/pythonlib: unified use of addonbase for mac

### PR DESCRIPTION
**Pull request by nilason**
The main goal for this PR is to set default GRASS_ADDON_BASE for macOS to reflect the settings of
GRASS app bundle as seen in e.g. [macosx/app/grass.sh.in](https://github.com/OSGeo/grass/blob/c963fbe0c429c5e6d36c55768473690f6345bbbd/macosx/app/grass.sh.in).
Historically GRASS_ADDON_BASE in the Mac bundle is set to:
`$HOME/Library/GRASS/[MAJORVERSION].[MINORVERSION]/Modules/`
e.g. `$HOME/Library/GRASS/7.9/Modules/`.

This is the file structure using GRASS mac bundle, up to this date:
```
$HOME/
├── .grass7/
├── .grass8/
│   ├── rc
│   ├── toolboxes/
│   └── wx.json
└── Library/
    └── GRASS/
        ├── 7.8/
        ├── 7.9/
        └── 8.0/
            └── Modules/
                ├── addons_paths.json
                ├── bin/
                ├── docs/
                ├── etc/
                ├── extensions.xml
                ├── lib/
                ├── modules.xml
                └── scripts/
```

Initially I propose to keep this, with the minor change of "Modules" to "Addons":
`$HOME/Library/GRASS/[MAJORVERSION].[MINORVERSION]/Addons/`

Along with above changes I also addressed the following related issues:
- for pythonlib grass.script.setup: use GRASS_ADDON_BASE if set, else fallback to defaults for each OS
- replaces hardcoded GRASS version as part file/dir names with a generated/dynamic version
- simplified platform detection code in modified files

### CODE REVIEW

**Review by wenzeslaus — CHANGES_REQUESTED** [review]
This seems to be a general improvement and definitively the code should reflect whatever is practically used on a given platform. I have only minor comments, although it is hard to actually make sure that both implementations are equal. However, moving all this to a library is for another PR which now hopefully will be more straightforward.

I'm not overly excited about the version global variables. I think these eventually will need to be a function which will be able to fall back to using `g.version -g` or `grass --version` in case the variables were not replaced (or the other way around); possibly also issuing a warning if the versions do not match. Anyway, outside of this PR.

File: lib/init/grass.py [hunk 1]
```diff
@@ -490,10 +492,9 @@ def create_tmp(user, gis_lock):
     if not tmp:
         tmp = tempfile.gettempdir()
 
+    tmpdir_name = f"grass{GRASS_VERSION_MAJOR}-{user!s}-{gis_lock!s}"
```
[the remark this answers is not in the document]
Comment (nilason) [reply lib/init/grass.py]: > Is the !s needed here?

Not at all.

File: lib/init/grass.py [hunk 2]
```diff
@@ -92,13 +92,15 @@
...
+GRASS_VERSION_MAJOR = "@GRASS_VERSION_MAJOR@"
+GRASS_VERSION_MINOR = "@GRASS_VERSION_MINOR@"
 LD_LIBRARY_PATH_VAR = "@LD_LIBRARY_PATH_VAR@"
 CONFIG_PROJSHARE = os.environ.get("GRASS_PROJSHARE", "@CONFIG_PROJSHARE@")
 
 # Get the system name
 WINDOWS = sys.platform == "win32"
 CYGWIN = "cygwin" in sys.platform
-MACOSX = "darwin" in sys.platform
+MACOS = "darwin" in sys.platform
```
[the remark this answers is not in the document]
Comment (nilason) [reply lib/init/grass.py]: Updated both `lib/init/grass.py` and `script/setup.py` to use `sys.platform.startswith()`.

**Comment by nilason** [comment]
@cmbarton 
Do you have any objection to rename the directory for addons (starting with G8):

old:
`$HOME/Library/GRASS/8.0/Modules/`

new:
`$HOME/Library/GRASS/8.0/Addons/`

This is to (1) make it more consistent with *nix and win, which both uses the dir name `addons`; and (2) this more clearly defines the content (there are modules in core base too, but only here there are addons).

**Comment by cmbarton** [comment]
> @cmbarton Do you have any objection to rename the directory for addons (starting with G8):
> 
> old: `$HOME/Library/GRASS/8.0/Modules/`
> 
> new: `$HOME/Library/GRASS/8.0/Addons/`
> 
> This is to (1) make it more consistent with *nix and win, which both uses the dir name `addons`; and (2) this more clearly defines the content (there are modules in core base too, but only here there are addons).

Sorry for the slow reply. I was in Washington, DC to coordinate a meeting of international modeling orgs. My first plane trip in nearly 2 years. 

This seems like a good idea and will be more understandable going forward. The previous approach simply mirrored the directory structure of GRASS binaries. So it was sometimes difficult to find where things were supposed to be put (especially before g.extension got better at automatically placing things). The main thing is this needs to get into the search path for a GRASS 8 execution environment.

**Review by wenzeslaus — APPROVED** [review]

**Comment by nilason** [comment]
Merged.
@wenzeslaus @cmbarton Thank you!

@wenzeslaus or @neteler : Could you please backport this, as I have no access to my dev machine the next few days.

**Comment by neteler** [comment]
Sure, backported.

**Comment by nilason** [comment]
> Sure, backported.

Thank you!

**Pull request merged**